django_auth_adfs/backend.py

```diff
@@ -172,22 +173,25 @@ def update_user_attributes(self, user, claims):
 
         required_fields = [field.name for field in user._meta.fields if field.blank is False]
 
-        for field, claim in settings.CLAIM_MAPPING.items():
-            if hasattr(user, field):
-                if claim in claims:
-                    setattr(user, field, claims[claim])
-                    logger.debug("Attribute '%s' for user '%s' was set to '%s'.", field, user, claims[claim])
-                else:
-                    if field in required_fields:
-                        msg = "Claim not found in access token: '{}'. Check ADFS claims mapping."
-                        raise ImproperlyConfigured(msg.format(claim))
+        for field, claim in mapped_fields.items():
+            if type(claim) != dict:
```

Comment:
You should use `isinstance(claim, dict)` so that it'll support other types of dict-like objects.

Reply:
Done: https://github.com/snok/django-auth-adfs/pull/242/commits/91f7ee5273cacda41e7a790e378bd8698fc3f39a